### PR TITLE
fix (cherry-pick): Remove old What's New messages (#25048) for 12.0.0

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Removes old what's new items and prevents the What's New modal from displaying.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25157?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Open extension
2. Don't see what's new

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/whats-new-popup/whats-new-popup.js [hunk 1]
```diff
@@ -9,21 +9,13 @@ import {
   MetaMetricsEventName,
 } from '../../../../shared/constants/metametrics';
 import {
-  ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
-  NOTIFICATION_BLOCKAID_DEFAULT,
```
Comment (georgewrmarshall) [inline ui/components/app/whats-new-popup/whats-new-popup.js:13]: Should all instances of this const be removed from the cod base? It's still present in tests

Screenshot 2024-06-11 at 8 39 05 AM

File: ui/components/app/whats-new-popup/whats-new-popup.js [hunk 2]
```diff
@@ -33,38 +25,11 @@ import { updateViewedNotifications } from '../../../store/actions';
...
-      updateViewedNotifications({ 24: true });
-    },
     [NOTIFICATION_DROP_LEDGER_FIREFOX]: () => {
       updateViewedNotifications({ [NOTIFICATION_DROP_LEDGER_FIREFOX]: true });
     },
-    [NOTIFICATION_U2F_LEDGER_LIVE]: () => {
-      updateViewedNotifications({ [NOTIFICATION_U2F_LEDGER_LIVE]: true });
-    },
-    ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
-    [NOTIFICATION_BLOCKAID_DEFAULT]: () => {
```
Comment (georgewrmarshall) [inline ui/components/app/whats-new-popup/whats-new-popup.js:58]: Still present in tests

Screenshot 2024-06-11 at 8 40 14 AM

**Review by georgewrmarshall — COMMENTED** [review]
I don't have full context of this PR but looking good left a couple comments:
- Confirmed What's New Modal doesn't appear in fresh install of the extension ✅ 
- Checked no instances of removed local keys from `en` remain ✅ 
- Checked no instances of removed consts remain ❌

File: shared/notifications/index.ts [hunk 3]
```diff
@@ -5,10 +5,6 @@
  * into numbers in only one place. This should make merge conflicts easier.
  */
 export const NOTIFICATION_DROP_LEDGER_FIREFOX = 25;
-export const NOTIFICATION_U2F_LEDGER_LIVE = 28;
-export const NOTIFICATION_BLOCKAID_DEFAULT = 29;
```
Comment (georgewrmarshall) [inline shared/notifications/index.ts:9]: Should all instances of this const be removed from the cod base? It's still present in tests

Screenshot 2024-06-11 at 8 39 05 AM

Comment (danjm) [reply shared/notifications/index.ts:9]: nice find. These should be fixed on the develop branch

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🎸

**Review by NidhiKJha — APPROVED** [review]
LGTM

**Pull request merged**